lib/created_message_queue_handler.ts

```diff
@@ -245,183 +218,187 @@ export async function handleMessage(
     return left(
       PermanentError(
         `Fiscal code has no associated email address and no default email address was provided|${
-          newMessageWithoutContent.fiscalCode
+          newMessageWithContent.fiscalCode
         }`
       )
     );
   }
 
   const newNotification = createNewNotification(
     ulidGenerator,
-
-    newMessageWithoutContent.fiscalCode,
-    newMessageWithoutContent.id
+    newMessageWithContent.fiscalCode,
+    newMessageWithContent.id
   );
 
-  return saveNotification(notificationModel, {
-    ...newNotification,
-    channels: {
-      [NotificationChannelEnum.EMAIL]: maybeNotificationChannelEmail.toUndefined()
-    }
-  });
-}
+  const errorOrNotification = await notificationModel.create(
+    {
+      ...newNotification,
+      channels: {
+        [NotificationChannelEnum.EMAIL]: maybeNotificationChannelEmail.toUndefined()
+      }
+    },
+    newNotification.messageId
+  );
 
-export function processResolve(
-  errorOrNotification: Either<RuntimeError, RetrievedNotification>,
-  context: ContextWithBindings,
-  messageContent: MessageContent,
-  senderMetadata: CreatedMessageEventSenderMetadata
-): void {
   if (isLeft(errorOrNotification)) {
-    const error = errorOrNotification.value;
-    if (isTransient(error)) {
-      winston.error(
-        `CreatedMessageQueueHandler|Transient error|${error.message}`
-      );
-      // schedule a retry in case of transient errors
-      // retry function is async and calls context.done() by itself
-      // TODO: update message status (PROCESSING)
-      const queueService = createQueueService(queueConnectionString);
-      return retryMessageEnqueue(queueService, MESSAGE_QUEUE_NAME, context);
-    } else {
-      // the message processing failed with an unrecoverable error
-      // TODO: update message status (FAILED)
-      winston.error(
-        `CreatedMessageQueueHandler|Permanent error|${error.message}`
-      );
-      return context.done();
-    }
+    return left(TransientError("Cannot save notification to database"));
   }
 
-  const notification = errorOrNotification.value;
+  return right(errorOrNotification.value);
+}
 
+export function processGenericError(error: Error): void {
```

Comment:
logic here is we can 

- fail with a transient error -> trigger a retry
  -> check if the message has reached max number of retries -> update status
- fail with a permanent error -> stop message processing -> update status
- fail with some unexpected error (outside handleNotification) -> stop message processing -> update status
- success -> update status -> output bindings

Reply:
ditto